library/gd_mf_weights.cc

```diff
@@ -58,8 +58,23 @@ int main(int argc, char *argv[])
   size_t rank = atoi(location);
 
   // global model params
-  unsigned char left_ns = model->pairs[0][0];
-  unsigned char right_ns = model->pairs[0][1];
+  string firstPair;
+  for (auto const& i : model->interactions)
```

Comment:
You can use find_first_of https://en.cppreference.com/w/cpp/algorithm/find_first_of